lib/vm/src/trap/traphandlers.rs

```diff
@@ -153,13 +159,60 @@ cfg_if::cfg_if! {
                 }
                 _ => None,
             };
-
+            let trap_code = match signum {
+                // check if it was cased by a UD and if the Trap info is a payload to it
+                libc::SIGILL => {
+                    let mut val: u8 = 0;
+                    if cfg!(target_arch = "x86_64") {
+                        let addr = (*siginfo).si_addr() as usize;
+                        val = if read_unaligned(addr as *mut u8)&0xf0 == 0x40
+                              && read_unaligned((addr+1) as *mut u8) == 0x0f
+                              && read_unaligned((addr+2) as *mut u8) == 0xb9 {
+                                read_unaligned((addr+3) as *mut u8)
+                            } else if read_unaligned(addr as *mut u8) == 0x0f
+                                    && read_unaligned((addr+1) as *mut u8) == 0xb9 {
+                                read_unaligned((addr+2) as *mut u8)
+                            } else {
+                                0
+                            }
+                    }
+                    if cfg!(target_arch = "aarch64") {
+                        let addr = (*siginfo).si_addr() as usize;
+                        val = if read_unaligned(addr as *mut u32)&0xffff0000 == 0 {
+                            read_unaligned(addr as *mut u8)
+                        } else {
+                            0
+                        }
+                    }
+                    if val&MAGIC == MAGIC {
```

Comment:
And use the `.and_then` method on `Option` here.

Reply:
and done